examples/vision/semantic_segmentation_deeplab_v3.py

```diff
@@ -0,0 +1,340 @@
+
+!pip install gdown
+
+# This sample uses Keras Core, the multi-backend version of Keras.
+# The selected backend is TensorFlow (other supported backends are 'jax' and 'torch')
+import os
+os.environ['KERAS_BACKEND'] = 'tensorflow'
+"""
+
+import os
+import cv2
+import numpy as np
+from glob import glob
+import matplotlib.pyplot as plt
+import tensorflow as tf
+
+
+import keras_core as keras
+
+import keras_cv
+
+from keras_cv.models import ResNet18V2Backbone, DeepLabV3Plus
+
+print("TensorFlow version:", tf.__version__)
+print("KerasCV version:", keras_cv.__version__)
+
+"""
+## Accelerator
+
+Detect hardware, return appropriate distribution strategy
+"""
+
+try:
+    # detect and init the TPU
+    tpu = tf.distribute.cluster_resolver.TPUClusterResolver.connect()
+    # instantiate a distribution strategy
+    strategy = tf.distribute.TPUStrategy(tpu)
+except ValueError:
+    print("TPU not activated")
+    strategy = (
+        tf.distribute.MirroredStrategy()
+    )  # Works on CPU, single GPU and multiple GPUs in a single VM.
+
+print("replicas:", strategy.num_replicas_in_sync)
+
+"""
+## Loading the data into TensorFlow Datasets
+
+Training on the entire HuBMAP - Hacking the Human Vasculature dataset with more than 7000
+images takes a lot of time, hence we will be using
+a smaller subset of about 400 images for training our model in this example.
+"""
+
+"""
+The dataset is an extract of a dataset used in a Kaggle competition.
+
+You can find more information
+[here](https://www.kaggle.com/datasets/alexia/hubmap-kidney).
+
+
+
+"""
+
+"""shell
+!gdown 1HGYBgRP362lW5Xtt-RW3mYNibjd45r8P
+!unzip hubmap-kidney.zip -d hubmap-kidney
+"""
+
+DATA_DIR = "hubmap-kidney/"
+IMG_DIR = DATA_DIR + "img/"
+MASK_DIR = DATA_DIR + "mask/"
+
+import os
+
+images_list = []
+for dir_path, dir_names, file_names in os.walk(IMG_DIR):
+    for file in file_names:
+        if file.endswith("png"):
+            images_list.append(file)
+
+
+NUM_IMAGES = 416
+IMG_HEIGHT = 512
+IMG_WIDTH = 512
+# Define the split percentages
+TRAIN_SPLIT = 0.8
+NUM_TRAIN_IMAGES = int(NUM_IMAGES * TRAIN_SPLIT)
+
+BATCH_SIZE = 4
+
+train_img_list = images_list[:NUM_TRAIN_IMAGES]
+val_img_list = images_list[NUM_TRAIN_IMAGES:]
+
+
+def gen_pairs(img_list):
+    def gen():
+        for img_id in img_list:
+            image = tf.io.read_file(IMG_DIR + "/" + img_id)
+            image = tf.image.decode_png(image, channels=3)
+            image.set_shape([IMG_HEIGHT, IMG_WIDTH, 3])
+            mask = tf.io.read_file(MASK_DIR + "/" + img_id)
+            mask = tf.image.decode_png(mask, channels=1)
+            mask = tf.one_hot(tf.squeeze(mask), 2)
+            mask.set_shape([IMG_HEIGHT, IMG_WIDTH, 2])
+            yield (image, mask)
+
+    return gen
+
+
+# Prepare the training dataset.
+train_gen = tf.data.Dataset.from_generator(
+    generator=gen_pairs(train_img_list),
+    output_signature=(
+        tf.TensorSpec(shape=(IMG_WIDTH, IMG_HEIGHT, 3), dtype=tf.uint8),
```

Comment:
Not critical here, since images are square, but this should be IMG_HEIGHT, IMG_WIDTH (order reversed)